akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -16,7 +16,7 @@ import akka.actor.typed.Behavior.unhandled
 import BehaviorBuilder._
 
 /**
```

Comment:
we use
```
import akka.japi.function.{ Function, Function2, Predicate }
```

I think that is confusing. Function and Function2 exists in scala. I'd rather rename those imports to JFunction for clarity.

Related question is if we have a good reason for using japi here but `java.util.function` in persistence?

Reply:
I never quite understood why we use `japi` ones, something about scala version expanding lambas compatibility something?

I agree if there is no reason we should use the built in Java types

Reply:
isn't it something with variance too? I think we introduced the new function types because of that in Streams.

since the typical usage is to delegate to a method we could try if it makes any difference for combinations of things like this:
```
.onMessage(Orange.class, this::receiveFruit)
.onMessage(Banana.class, this::receiveFruit)
.onMessage(Fruit.class, this::receiveFruit)

private void receiveFruit(Orange o);
private void receiveFruit(Banana o);
private void receiveFruit(Fruit f);
```

Reply:
Yes, you are right, variance as well. The java ones are invariant since they are defined in Java.